src/typechecker.erl

```diff
@@ -1658,6 +1658,10 @@ do_type_check_expr(Env, {call, _, {atom, _, TypeOp}, [Expr, {string, _, TypeStr}
 do_type_check_expr(Env, {call, _, {atom, _, record_info}, [_, _]} = Call) ->
     Ty = get_record_info_type(Call, Env),
     {Ty, Env, constraints:empty()};
+do_type_check_expr(Env, {call, P, {remote, _, _Mod, {atom, _, module_info}} = Name, Args}) ->
+    Arity = arity(length(Args)),
+    FunTy = get_module_info_type(Arity),
```

Comment:
I suppose we have a problem if Arity > 1. Perhaps we should break this clause up into explicit matching on arity 0 (`Args = []`) and 1 (`Args = [_]`) and fall though for any other arity?

Reply:
Changed in #500.